### ISSUE TITLE
What can use static assets from RCLs?

### ISSUE THREAD

**Issue by thatnerdyguy** [aspnet/AspNetCore.Docs#12884]
This page says:
"In ASP.NET Core 3.0 Preview, Razor class libraries aren't compatible with Blazor client-side apps."
and
"In ASP.NET Core 3.0 Preview, Blazor server-side apps can't consume static assets from either a Razor class library (razorclasslib) or a Blazor library (blazorlib)."
but later
"ASP.NET Core allows creating RCLs that include static assets that are available to a consuming app."

But ... if client-side Blazor can't use RCLs and server-side Blazor can't consume static assets from an RCL... what exactly can use them?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 37b02665-c8ab-bd91-8057-3b07b4e238d3
* Version Independent ID: e4f176f5-0b94-d532-53d5-cffce437961a
* Content: [Razor components class libraries](https://docs.microsoft.com/en-us/aspnet/core/blazor/class-libraries?view=aspnetcore-3.0&tabs=visual-studio)
* Content Source: [aspnetcore/blazor/class-libraries.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/class-libraries.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Thanks for the note @thatnerdyguy ... we just had a software release for Pre6 two days ago. I need to perform some touch-ups to those remarks. I'll try to take a look tomorrow morning (Saturday) and ping u back.

**Comment by guardrex** [comment]
@danroth27 There are some issues to address ...

* The Component Library topic wasn't fully updated: As @thatnerdyguy calls out here, we should pull the remarks that state RCL can't have static assets.
* `blazorlib` is still a template: Tea leaves say that it will be around until Pre9 looking at [Remove the Blazor Class Library template (aspnet/AspNetCore \#10721)](https://github.com/aspnet/AspNetCore/issues/10721). btw - The goal of getting client-side on RCL (either tracking on that engineering issue or another one) isn't on the [Blazor Roadmap (aspnet/AspNetCore \#8177)](https://github.com/aspnet/AspNetCore/issues/8177), which doesn't list Pre9.
* Javier [called out on the PR (aspnet/AspNetCore.Docs \#12722)](https://github.com/aspnet/AspNetCore.Docs/pull/12722#discussion_r290945120) that RCL is an ASP.NET Core scenario, not just Blazor.
  * Adding the static asset content to the Blazor Component topic is fine because this topic is just telling the reader how to use an RCL as a component lib.
  * However, the content wasn't cross-added (via an INCLUDE or just copied into) into the [Create reusable UI using the Razor Class Library project in ASP.NET Core](https://docs.microsoft.com/aspnet/core/razor-pages/ui-class?view=aspnetcore-3.0&tabs=visual-studio) topic.
* The Component Library topic doesn't cross-link to the "Reusable UI RCL" topic. I'll provide it as an additional resource and link the first reference to RCL in the topic.

TL;DR ... I guess I just need to know if the `blazorlib` template for client-side is still the correct approach until we get to Pre9? I think I have enough info on the rest to get a PR drafted.
